### PR TITLE
Add PermissionChangeObject and expose a Management realm getter

### LINKED ISSUE #955: Client Permissions
- [ ] `PermissionChange` class
- [ ] `User.OpenManagementRealm()` method

### PR DESCRIPTION
**Pull request by fealebenpae**
Closes #955

### CODE REVIEW

File: Shared/Realm.Sync.Shared/Permissions/IPermissionObject.cs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        string Id { get; set; }
+
+        /// <summary>
+        /// Gets or sets the creation time of this object.
+        /// </summary>
+        DateTimeOffset CreatedAt { get; set; }
+
+        /// <summary>
+        /// Gets or sets when the object was updated the last time.
+        /// This should be filled
```
Comment (nirinchev) [inline Shared/Realm.Sync.Shared/Permissions/IPermissionObject.cs]: What's with the newlines?

File: Platform.PCL/Realm.Sync.PCL/Permissions/PermissionChangePCL.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    /// <remarks>
+    /// <see cref="PermissionChange"/> objects allow to grant and revoke permissions by setting
+    /// <see cref="MayRead" />, <see cref="MayWrite" /> and <see cref="MayManage" /> accordingly. 
+    /// If any of these flags are not set, these are merged
+    /// with either the existing or default permissions as applicable. As a
+    /// side-effect this causes that the default permissions are permanently
+    /// materialized for the affected Realm files and the affected user.
+    /// </remarks>
+    [Preserve(AllMembers = true)]
+    public class PermissionChange : RealmObject, IPermissionObject
```
Comment (nirinchev) [inline Platform.PCL/Realm.Sync.PCL/Permissions/PermissionChangePCL.cs]: Should this be `[Explicit]`?

Comment (fealebenpae) [inline Platform.PCL/Realm.Sync.PCL/Permissions/PermissionChangePCL.cs]: We don't care about any attribute metadata on the classes in the PCL.  I should even get rid of the `Preserve` attribute.

Comment (nirinchev) [inline Platform.PCL/Realm.Sync.PCL/Permissions/PermissionChangePCL.cs]: Cool, I thought they might somehow leak into the assemblies our weaver processes, but if they don't, I guess we're safe :)

Comment (fealebenpae) [inline Platform.PCL/Realm.Sync.PCL/Permissions/PermissionChangePCL.cs]: The weaver doesn't care about the `Explicit` attribute.

Comment (nirinchev) [inline Platform.PCL/Realm.Sync.PCL/Permissions/PermissionChangePCL.cs]: Right, my bad.

**Pull request merged**